### PR TITLE
Create eslint rule to ban module.exports

### LINKED ISSUE #19257: Ban module.exports outside of node_modules/
Context: https://github.com/ampproject/amphtml/issues/18851#issuecomment-431731610

### PR DESCRIPTION
**Pull request by estherkim**
Bans `module.exports` from all directories except ads, build-system, testing, and validator

Fixes #19257

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "amphtml-internal/no-import-rename": 2,
     "amphtml-internal/no-is-amp-alt": 2,
     "amphtml-internal/no-mixed-operators": 2,
+    "amphtml-internal/no-module-exports": 2,
```
[the remark this answers is not in the document]
Comment (estherkim) [reply .eslintrc]: Luckily there was only 1 error! It was coming from `babel.config.js` which I added it to `.eslintignore`.

[the remark this answers is not in the document]
Comment (rsimha) [reply .eslintrc]: SGTM!

**Comment by rsimha** [comment]
@erwinmombay @jridgewell I'll merge this soon if you don't have any comments.

File: build-system/eslint-rules/no-module-exports.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    schema: [],
+    messages: {
+      unexpected: "Unexpected use of `module.exports` statement."
+    }
+  },
+
+  create(context) {
+
+    function isModuleExports(node) {
+      return node.object.type === 'Identifier' && (/^(module.exports)$/).test(node.object.name)
```
Comment (jridgewell) [inline build-system/eslint-rules/no-module-exports.js]: This is incorrect. `module.exports` is a `MemberExpression` composed of two `Identifiers`, not a plain `Identifier`. Please see https://astexplorer.net/#/gist/f81b69fca8b30a5592b4ef13cde5fff8/096f72d923b5aaa0e7d1470037bdc2403af177f5, which is essentially a ESLint repl to help build plugins.

Comment (estherkim) [reply build-system/eslint-rules/no-module-exports.js]: Got it, thanks for the link. Submitted fix #19581

File: build-system/eslint-rules/no-module-exports.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  create(context) {
+
+    function isModuleExports(node) {
+      return node.object.type === 'Identifier' && (/^(module|exports)$/).test(node.object.name)
+    }
+
+    return {
+      MemberExpression(node) {
+        if (isModuleExports(node)) {
+            context.report({ node, messageId: "unexpected"});
```
Comment (danilo25254) [inline build-system/eslint-rules/no-module-exports.js]: ``

**Pull request merged**